test/common/http/http2/http2_frame.cc

```diff
@@ -157,6 +157,38 @@ Http2Frame Http2Frame::makePriorityFrame(uint32_t stream_index, uint32_t depende
   return frame;
 }
 
+Http2Frame Http2Frame::makeEmptyPushPromiseFrame(uint32_t stream_index,
+                                                 uint32_t promised_stream_index,
+                                                 HeadersFlags flags) {
+  static constexpr size_t kEmptyPushPromisePayloadSize = 4;
+  Http2Frame frame;
+  frame.buildHeader(Type::PushPromise, kEmptyPushPromisePayloadSize, static_cast<uint8_t>(flags),
+                    makeRequestStreamId(stream_index));
+  uint32_t promised_stream_id = makeRequestStreamId(promised_stream_index);
+  memcpy(&frame.data_[HeaderSize], reinterpret_cast<void*>(&promised_stream_id), sizeof(uint32_t));
```

Comment:
This seems a bit fragile, I think in http2_frame.cc we do this octet by octet instead. Probably OK as test code, but anytime I see memcpy my spidey sense is activated (see https://github.com/envoyproxy/envoy/issues/9328).

Reply:
Added asserts to the test.
It seems that a better approach would be following [abseil's span suggestion](https://abseil.io/tips/93) but it requires handling different types.
If required I can add that (if it wasn't test code I would push for this).